### PR TITLE
Only show Suggest LOI job cards when zoomed in

### LINKED ISSUE #1800: Only show "collect data at new site/LOI" / suggest LOI card at same zoom level as other cards
We should show a hint when zoomed out and cards are hidden akin.

@vittorino @amegantz @gino-m For UX and copy.

### PR DESCRIPTION
**Pull request by JSunde**
Fixes #1800

### CODE REVIEW

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt [hunk 1]
```diff
@@ -122,8 +124,10 @@ internal constructor(
         .toLiveData()
 
     suggestLoiJobs =
-      surveyRepository.activeSurveyFlow.map {
-        it?.jobs?.filter { job -> job.suggestLoiTaskType != null }?.toList() ?: listOf()
+      surveyRepository.activeSurveyFlow.combine(cameraZoomUpdates.asFlow()) { survey, zoomLevel ->
+        if (zoomLevel < CLUSTERING_ZOOM_THRESHOLD) return@combine listOf()
+       
+        survey?.jobs?.filter { job -> job.suggestLoiTaskType != null }?.toList() ?: listOf()
```
Comment (shobhitagarwal1612) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:130]: Will this refresh the list on every zoom level change? Or does it work like `distinctUntilChanged()` in Rx?

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt [hunk 2]
```diff
@@ -122,8 +124,10 @@ internal constructor(
         .toLiveData()
 
     suggestLoiJobs =
-      surveyRepository.activeSurveyFlow.map {
-        it?.jobs?.filter { job -> job.suggestLoiTaskType != null }?.toList() ?: listOf()
+      surveyRepository.activeSurveyFlow.combine(cameraZoomUpdates.asFlow()) { survey, zoomLevel ->
```
Comment (gino-m) [inline ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:127]: Can this be rewritten using Coroutines/Flow?

**Review by gino-m — CHANGES_REQUESTED** [review]

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt (hunk 2 above)
Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:127]: This is using Flow

Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:127]: suggestLoiJobs is a Flow<List<Job>>

Comment (gino-m) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt:127]: Oops sorry I missed that!

File: ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt (hunk 1 above)
Comment (shobhitagarwal1612) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt]: I don't see any change. Did you forget to push it?

Comment (JSunde) [reply ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt]: Weird I wasn't even seeing it locally, but I swear I made the change. Anyways I made the change again and pushed it now!

**Review by shobhitagarwal1612 — APPROVED** [review]

**Pull request merged**